news/2019-02-07-weld-310Final.asciidoc

```diff
@@ -0,0 +1,81 @@
+---
+layout: news
+title: Weld 3.1.0.Final
+author: Matej Novotny
+priority: 1
+change_frequency: weekly
+excerpt: Weld 3.1.0.Final released
+desc: Weld 3.1.0.Final released
+tags: [release]
+---
+:linkattrs:
+
+Weld 3.1.0.Final and Weld API 3.1.Final are now up for grabs. What's in it for you?
+
+There are quite a few new things in the mix - `InterceptionFactory` improvements, new API clases and methods, improvements to class defining for integrators in preparation for JDK 12.
+We now also support CDI context propagation between threads; there is a whole new SPI that allows users and/or frameworks to propagate request, session or conversation contexts.
+Last but no least, there are bug fixes, so let's get right into it!
+
+NOTE: Integrators (WildFly, Liberty, GlassFish, ...) will have to adjust to the changes made in SPI. Those are, most notably, the removal of long deprecated methods (link:https://issues.jboss.org/browse/WELD-2558[WELD-2558, window="_blank"]) and reworked `ProxyServices` which now delegate class defining responsibility to the integrator (link:https://issues.jboss.org/browse/WELD-2556[WELD-2556, window="_blank"]). This is crucial for Weld to operate on newer JDKs (12) and while Weld will work with the old approach for now, we will eventually fully swap to this new SPI. 
+
+Fixes and improvements:
+
+* Weld API/SPI
+** Added API to allow for CDI context propagation between threads (link:https://issues.jboss.org/browse/WELD-2497[WELD-2497, window="_blank"])
+*** New API class was introduced (`WeldAlterableContext`), this class offers methods to manipulate context state
+*** You can now propagate `Request`, `Session` and `Conversation` contexts between threads as they all implement `WeldAlterableContext`
+*** This comes with certain limitations is mostly designed for frameworks which would do this for you (such as MicroProfile Concurrency) but anyone can use it
```

Comment:
This comes with certain limitations is mostly designed for frameworks... - the wording does not look correct to me, or I just don't understand